### PR TITLE
test(driver-adapters): unify Driver Adapters assertions in `issues/15204` when testing with LibSQL

### PR DESCRIPTION
**Pull request by jkomyno**
This PR contributes to https://github.com/prisma/team-orm/issues/514.

It lets issues/15204 test suite pass, albeit it demonstrates more differences in error messages than originally expected.

Consider the following schema portion:

```prisma
model TestModel {
  id String @id
  field Int | BigInt // switch `Int` with `BigInt` depending on the test scenario
}
```

After create an entry in `TestModel` with a raw query like

```sql
INSERT INTO "TestModel" ("id", "field") VALUES ("1", 1.84467440724388e+19)
```

and attempting to retrieve such entry via 

```typescript
await prisma.testModel.findMany()
```

would result in 3 different kinds of error objects depending on the situation, all with a field `code: 'P2023'`.

1. Using Rust drivers: the error object has a field `message` saying "Inconsistent column data: Could not convert from `BigDecimal(18446744072438800000)` to `Int | BigInt`".
2. Using LibSQL Driver Adapter + `Int` as the field type: the error object has a field `message` saying "Conversion failed: number must be an integer in column 'field'".
3. Using LibSQL Driver Adapter + `BigInt` as the field type: the error object has a field `message` saying "Inconsistent column data: Conversion failed: number must be an i64"

This PR, paired with https://github.com/prisma/prisma-engines/pull/4429, provides the same error message in case of `Int` and `BigInt` field types. That engines PR has also be used locally to ensure `pnpm run test:functional --flavor=js_libsql issues/15204` succeeds with this PR.

Note: while this issue emerged when testing `libsql`, the fixes are not `libsql` specific.

### CODE REVIEW

File: packages/client/tests/functional/issues/15204-conversion-error/tests.ts [hunk 1]
```diff
@@ -7,20 +6,29 @@ declare let prisma: PrismaClient
...
+              `Inconsistent column data: Could not convert from \`BigDecimal(18446744072438800000)\` to \`${fieldType}\``,
+            ),
+          }),
+        )
+      } else {
+        await expect(prisma.testModel.findMany()).rejects.toThrow(
+          expect.objectContaining({
+            code: 'P2023',
+            message: expect.stringContaining(
+              `Conversion failed: number must be an integer in column 'field', got '1.84467440724388e19'`,
```
[the remark this answers is not in the document]
Comment (jkomyno) [reply packages/client/tests/functional/issues/15204-conversion-error/tests.ts:27]: This string is coming from the`driver_adapters::proxy::js_value_to_quaint` function, which is Driver Adapters-specific.
The other message comes from a similar function that is Quaint-specific: [`sql_query_connector::row::row_value_to_prisma_value`](https://github.com/prisma/prisma-engines/blob/6e26301fe272ba4ba0598fe43eb5d8df030be4db/query-engine/connectors/sql-query-connector/src/row.rs#L241-L250).

[the remark this answers is not in the document]
Comment (janpio) [reply packages/client/tests/functional/issues/15204-conversion-error/tests.ts:27]: Is that then not a bug that should be fixed?
(I know this is an edge case, but the less special cases we have to cover the better)

[the remark this answers is not in the document]
Comment (jkomyno) [reply packages/client/tests/functional/issues/15204-conversion-error/tests.ts:27]: Not a bug to be fixed, as discussed in a previous call with Miguel and Alexey.
The fewer logic cases the better, I agree, but JS driver adapters inherently have different data type conversion than the other drivers (abstracted by the `quaint` + `sql_query_connector` crates). So no actionable change is available here, beyond https://github.com/prisma/prisma-engines/pull/4429 (which unifies control logic for two integer cases of `driver-adapters`, simplifying the test suite in this PR).

**Review by miguelff — APPROVED** [review]

**Review by aqrln — APPROVED** [review]

File: packages/client/tests/functional/issues/15204-conversion-error/tests.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (aqrln) [reply packages/client/tests/functional/issues/15204-conversion-error/tests.ts:27]: Additionally, the error message by the driver adapters is a bit more user-friendly and understandable in this case, so I'd prefer not to dumb it down even if we could do that.

Comment (jkomyno) [inline packages/client/tests/functional/issues/15204-conversion-error/tests.ts:27]:
```suggestion
              `Conversion failed: number must be an integer in column 'field', got 1.84467440724388e19`,
```

File: packages/client/tests/functional/issues/15204-conversion-error/tests.ts [hunk 2]
```diff
@@ -7,20 +6,29 @@ declare let prisma: PrismaClient
...
+              `Inconsistent column data: Could not convert from \`BigDecimal(18446744072438800000)\` to \`${fieldType}\``,
+            ),
+          }),
+        )
+      } else {
+        await expect(prisma.testModel.findMany()).rejects.toThrow(
+          expect.objectContaining({
+            code: 'P2023',
+            message: expect.stringContaining(
+              `Conversion failed: number must be an integer in column 'field', got 1.84467440724388e19`,
```
Comment (jkomyno) [inline packages/client/tests/functional/issues/15204-conversion-error/tests.ts:27]: Proposal: add back the quotes wrapping the value to `prisma-engines` once the release is done. Will open an issue/PR for this.

**Comment by Jolg42** [comment]
It appears one test is failing at the moment in `Tests / Client (func/adapters) (js_libsql, 1/6, 20)`
https://github.com/prisma/prisma/actions/runs/6855154888/job/18656855544?pr=21849#step:6:293
```
   ● issues.15204-conversion-error (provider=sqlite, fieldType=BigInt, js_libsql) › should return a descriptive error

    expect(received).rejects.toThrow(expected)

    Expected asymmetric matcher: ObjectContaining {"code": "P2023", "message": StringContaining "Conversion failed: number must be an integer in column 'field', got 1.84467440724388e19"}

    Received name:    "PrismaClientKnownRequestError"
    Received message: "
    Invalid `expect(prisma.testModel.findMany()` invocation in
    /home/runner/work/prisma/prisma/packages/client/tests/functional/issues/15204-conversion-error/tests.ts:23:39·
      20     }),
      21   )
      22 } else {
    → 23   await expect(prisma.testModel.findMany(
    Inconsistent column data: Conversion failed: number must be an integer in column 'field', got '1.84467440724388e19'"

          211 |     // TODO: Do request with callsite instead, so we don't need to rethrow
          212 |     if (error.code) {
        > 213 |       throw new PrismaClientKnownRequestError(message, {
              |             ^
          214 |         code: error.code,
          215 |         clientVersion: this.client._clientVersion,
          216 |         meta: error.meta,
```

**Comment by jkomyno** [comment]
For context: yesterday I've pushed commit https://github.com/prisma/prisma/pull/21849/commits/778d95398ceb2e0a9d1714c464027f2b1ad1cb39 to fix [this CI error](https://github.com/prisma/prisma/actions/runs/6855154888/job/18656855544) (the test used `'` quotes, the runtime didn't have them). Now, I'm reverting that change as the engine has changed and now `'` quotes are effectively needed, as previously expected.

File: packages/client/tests/functional/issues/15204-conversion-error/tests.ts (hunk 2 above)
Comment (jkomyno) [reply packages/client/tests/functional/issues/15204-conversion-error/tests.ts]: (No longer needed)

**Comment by jkomyno** [comment]
Local runs:

- `❯ pnpm run test:functional --flavor=js_libsql issues/15204`
  ```sh
   PASS  tests/functional/issues/15204-conversion-error/tests.ts
    issues.15204-conversion-error (provider=sqlite, fieldType=Int)
      ○ skipped should return a descriptive error
    issues.15204-conversion-error (provider=sqlite, fieldType=Int, js_libsql)
      ✓ should return a descriptive error (123 ms)
    issues.15204-conversion-error (provider=sqlite, fieldType=BigInt)
      ○ skipped should return a descriptive error
    issues.15204-conversion-error (provider=sqlite, fieldType=BigInt, js_libsql)
      ✓ should return a descriptive error (4 ms)
  ```

- `❯ pnpm run test:functional --provider=sqlite issues/15204`
  ```sh
   PASS  tests/functional/issues/15204-conversion-error/tests.ts
    issues.15204-conversion-error (provider=sqlite, fieldType=Int)
      ✓ should return a descriptive error (76 ms)
    issues.15204-conversion-error (provider=sqlite, fieldType=Int, js_libsql)
      ○ skipped should return a descriptive error
    issues.15204-conversion-error (provider=sqlite, fieldType=BigInt)
      ✓ should return a descriptive error (4 ms)
    issues.15204-conversion-error (provider=sqlite, fieldType=BigInt, js_libsql)
      ○ skipped should return a descriptive error
  ```

**Comment by jkomyno** [comment]
All `js_libsql` tests are passing, and the failing `js_planetscale` one is flaky. I'm merging this.
